src/server/utils/wallets/getLocalWallet.ts

```diff
@@ -74,11 +78,18 @@ export const getLocalWallet = async ({
 export const getLocalWalletAccount = async (
   walletAddress: Address,
 ): Promise<Account> => {
-  const json = await new LocalFileStorage(walletAddress).getItem("");
-  if (!json) {
-    throw new Error(`Wallet not found for address ${walletAddress}`);
+  const walletDetails = await getWalletDetails({ address: walletAddress });
+
+  if (
+    !walletDetails ||
+    walletDetails.type !== "local" ||
+    !walletDetails.encryptedJson
+  ) {
+    throw new Error(`Local Wallet not found for address ${walletAddress}`);
   }
 
+  const json = walletDetails.encryptedJson;
+
   const wallet = await Wallet.fromEncryptedJson(
```

Comment:
this can be moved over to using `decryptoJsonWallet` from `legacyLocalCrypto`. The interface isn't as clunky here though, so we can let it be too.